openraft/src/raft/impl_raft_blocking_write.rs

```diff
@@ -30,17 +30,19 @@ where C: RaftTypeConfig<Responder = OneshotResponder<C>>
     /// - It proposes a **joint** config.
     /// - When the **joint** config is committed, it proposes a uniform config.
     ///
-    /// If `retain` is true, then all the members which not exists in the new membership,
+    /// If `retain` is `true`, then all the members which not exists in the new membership,
     /// will be turned into learners, otherwise will be removed.
+    /// If `retain` is `false`, the removed voter will be removed from the cluster.
+    /// Existing learners will not be affected.
     ///
     /// Example of `retain` usage:
-    /// If the original membership is {"voter":{1,2,3}, "learners":{}}, and call
-    /// `change_membership` with `voters` {3,4,5}, then:
-    ///    - If `retain` is `true`, the committed new membership is {"voters":{3,4,5},
-    ///      "learners":{1,2}}.
-    ///    - Otherwise if `retain` is `false`, then the new membership is {"voters":{3,4,5},
-    ///      "learners":{}}, in which the voters not exists in the new membership just be removed
-    ///      from the cluster.
+    /// If the original membership is `{"voter":{1,2,3}, "learners":{1,2,3,4,5}}`,
```

Comment:
I am a newbie to raft, but can a node be a voter as well as a learner? 🤔

Reply:
No, If a node is allowed to grant a RequestVote RPC, it is voter. Otherwise it is a Learner, which just receive log replication.

Reply:
So is this document kinda incorrect? Node 1/2/3 are in both voter and learner list

Reply:
You are right. The doc is outdated. `learners` should be `nodes`, which includes voters and learners.

Reply:
Get it, thanks for the reply:)

Reply:
@SteveLauC 
I'm gonna fix this typo in:
- https://github.com/datafuselabs/openraft/pull/1199